### PR TITLE
DO NOT MERGE: doc: create troubleshooting guide

### PR DESCRIPTION
**Pull request by yizha1**
Update:
- create troubleshooting guide

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

File: content/en/docs/troubleshooting.md [hunk 1]
```diff
@@ -1,6 +1,88 @@
...
+
+## When I verify an artifact, I get the error 'Error: open $HOME/.config/notation/trustpolicy.json: no such file or directory'
+
+Users need to configure trust policies before verifying artifacts. The trust policies are configured in a file named `trustpolicy.json`, which is stored under notation configuration directory.
+
+For Linux, the notation configuration directory is `${HOME}/.config/notation/`. For macOS, the notation configuration directory is `${HOME}/Library/Application Support/notation/`. For Windows, the notation configuration folder is `%AppData%\Roaming\notation\`.
+
+Using CLI command to configure trust policies is planned in Notation future release. Before that, you can refer to this [guide](https://github.com/notaryproject/notation/blob/v1.0.0-rc.1/specs/commandline/verify.md#configure-trust-policy) to configure trust policies manually.
+
+## When I verify an artifact, I get the error 'ERRO authenticity validation failed. Failure reason: error while loading the trust store, "$HOME/.config/notation/truststore/x509/ca/mytruststore" does not exist'
```
Comment (FeynmanZhou) [inline content/en/docs/troubleshooting.md:34]: @yizha1  Can you shorten each heading and move the detailed problem description below it? It causes the heading and right sidebar layout to look not pretty on the [preview page](https://deploy-preview-125--notarydev.netlify.app/docs/troubleshooting/). See [Prometheus FAQ documentation](https://prometheus.io/docs/introduction/faq/) as a good sample.

image

**Review by FeynmanZhou — COMMENTED** [review]
Only a small suggestion.

File: content/en/docs/troubleshooting.md [hunk 2]
```diff
@@ -1,6 +1,88 @@
...
\ No newline at end of file
+---
+
+> NOTE:
+>
+> Suggest a troubleshooting item by raising a PR to our [documentation repository](https://github.com/notaryproject/notaryproject.dev)
+
+## Enabling notation CLI commands logging
+
+Printing log messages for debugging the problems can be enabled using the `--debug` or `-d` flag. Using the `--verbose` or `-v` flag to set verbosity level will be enough for most cases.
```
[the remark this answers is not in the document]
Comment (yizha1) [reply content/en/docs/troubleshooting.md:14]: @zr-msft it's about enabling verbose logging. I will fix the wording.

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by zr-msft — APPROVED** [review]

File: content/en/docs/troubleshooting.md [hunk 3]
```diff
@@ -1,6 +1,88 @@
...
+> Suggest a troubleshooting item by raising a PR to our [documentation repository](https://github.com/notaryproject/notaryproject.dev)
+
+## Enabling notation CLI commands logging
+
+Printing log messages for debugging the problems can be enabled using the `--debug` or `-d` flag. Using the `--verbose` or `-v` flag to enable verbose logging will be enough for most cases.
+
+For example:
+
+```shell
+# Enable the debugging logs
```
Comment (toddysm) [inline content/en/docs/troubleshooting.md:19]: What is the difference between debug and verbose? It does seem redundant although we print different information for each one. Clarifying when to use one or the other would be helpful.

File: content/en/docs/troubleshooting.md [hunk 4]
```diff
@@ -1,6 +1,88 @@
...
+Check the `Subject` info in the output. If it is the identity that signs the artifact, you need to add the `Subject` info into `trustedIdentities`.
+
+## I have configured trust policy, but I still get the error 'no applicable trust policy'
+
+This error indicates that the `registryScopes` property is not correctly configured. This property contains a list of repository URIs, where the artifacts are stored. You need to make sure the signing artifact is stored in one of the listed repositories. If not, you need to add the missing repository URI in `registryScopes`, or you can add a new trust policy for the missing repository.
+
+The repository URI is in the format of `${registry-name}/${namespace}/${repository-name}`. For example, if the artifact to be verified is `registry.acme-rockets.io/software/net-monitor@sha256:xxx`, then the value for `registryScopes` should be `registry.acme-rockets.io/software/net-monitor`, the following values are wrong
+
+- `registry.acme-rockets.io`
+- `registry.acme-rockets.io/software`
```
Comment (toddysm) [inline content/en/docs/troubleshooting.md:76]: As a clarification, we should add that wildcards are not supported, and users need to list each repository explicitly. This is not a very user-friendly experience for customers who can have thousands of repositories :(

File: content/en/docs/troubleshooting.md [hunk 5]
```diff
@@ -1,6 +1,88 @@
...
+
+## When I verify an artifact, I get the error 'malformed trustpolicy.json file'
+
+This is normally an encoding problem of `trustpolicy.json` file. Notation expects `utf-8 without BOM` or `ascii` encoding for `trustpolicy.json` file.
+
+For Windows user, Windows PowerShell (prior to v6) uses the Unicode `UTF-16LE` encoding by default, and `utf-8 without BOM` is not supported. If you are building `trustpolicy.json` file in Windows PowerShell (prior to v6), make sure you change the file encoding to `ascii`.
+
+## When I verify an artifact, I get the error 'Failed to unmarshal the payload content in the signature blob to envelope.Payload'
+
+This is normally an encoding problem of payload content in the signature envelope. Notary v2 signatures could be produced by different tools per [signature specification](https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.2/specs/signature-specification.md). The payload content is a `JSON` document defined in the signature specification, and the encoding should be `utf-8 without BOM` or `ascii`.
```
Comment (toddysm) [inline content/en/docs/troubleshooting.md:86]: Notary v2 should not be used anywhere in the documentation. We should change to "Notary signatures"

**Review by toddysm — CHANGES_REQUESTED** [review]

**Comment by yizha1** [comment]
Close this PR. Please review this one https://github.com/notaryproject/notaryproject.dev/pull/203

**Pull request closed without merging**